q2_longitudinal/_utilities.py

```diff
@@ -400,40 +391,84 @@ def _regplot_subplots_from_dataframe(state_column, metric, metadata,
 
 
 def _control_chart_subplots(state_column, metric, metadata, group_column,
-                            ci=95, palette='Set1', plot_control_limits=True,
-                            xtick_interval=None):
+                            individual_id_column, ci=95, palette='Set1',
+                            plot_control_limits=True, xtick_interval=None,
+                            yscale='linear', spaghetti='no'):
 
     groups = metadata[group_column].unique()
-    states = metadata[state_column].unique()
-    chart, axes = plt.subplots(len(groups) + 1, figsize=(6, 18))
+    states = sorted(metadata[state_column].unique())
+    fig_count = len(groups) + 1
+    fig_height = fig_count * 6
+    chart, axes = plt.subplots(fig_count, figsize=(6, fig_height))
 
     # determine x tick interval: autoscale so that ≤ 30 labels appear
     xtick_interval = _set_xtick_interval(xtick_interval, states)
 
-    # plot all groups together, compare variances
-    c, global_mean, global_std = _control_chart(
-        state_column, metric, metadata, group_column, ci=ci, palette=palette,
-        plot_control_limits=plot_control_limits, ax=axes[0],
-        xtick_interval=xtick_interval)
-    c.set_title('Group volatility comparison plot')
-    c = _set_xticks(c, metadata, state_column, states, xtick_interval)
-
     # plot individual groups' control charts
     colors = cycle(sns.color_palette(palette, n_colors=len(groups)))
-    for num in range(len(groups)):
-        group = groups[num]
-        group_md = metadata[metadata[group_column] == group]
+    cmap = {}
+    num = 1
+    for group, group_md in metadata.groupby(group_column):
+        color = next(colors)
+        cmap[group] = color
         c, gm, gs = _control_chart(
             state_column, metric, group_md, None, ci=ci, legend=False,
-            color=next(colors), plot_control_limits=True, ax=axes[num + 1],
+            color=color, plot_control_limits=plot_control_limits, ax=axes[num],
             palette=None, xtick_interval=xtick_interval)
         c.set_title('{0}: {1}'.format(group_column, group))
+        if spaghetti != 'no':
+            # plot group's sphaghetti on main plot and current subplot
+            for ax in [0, num]:
+                c = _make_spaghetti(
+                    group_md, state_column, metric, individual_id_column,
+                    states, ax=axes[ax], color=color, alpha=0.3,
```

Comment:
I guess this alpha would become adaptive someday?

Reply:
probably yes — I do not want to introduce too many parameters at this stage, though